### ISSUE TITLE
Can't update Magisk or install the standard version

### ISSUE THREAD

**Issue by leodexe** [phhusson/treble_experimentations#1368]
The recommended Magisk for this GSI https://github.com/phhusson/treble_experimentations/releases/download/v200/magisk-phh.zip
Cannot be upgraded as it will cause a bootloop, also installing any other version other than that will also fail and cause bootloop. Is there any workaround for this?
I'm using this GSI on my Moto Z2 Play https://sourceforge.net/projects/amber-gsi/files/PixelExperience_Plus/PixelExperience_Plus-10.0-arm64-a-gapps.img.xz/download
With this KVT installer https://androidfilehost.com/?fid=8889791610682867863

**Comment by MarcosPearce** [comment]
test these https://sourceforge.net/projects/expressluke-gsis/files/Extras/phh-magisk-ten/
should work in most cases

**Comment by leodexe** [comment]
Yes it worked, can't update to 20.4 as it still hangs on the logo but still it is better than nothing.

**Comment by PsyEng** [comment]
https://github.com/topjohnwu/Magisk/issues/844

**Comment by soredake** [comment]
https://t.me/c/1221084694/441
https://github.com/ExpressLuke/phh-magisk-builder/releases

**Comment by PsyEng** [comment]
@soredake Do I see it correct, that you´re scripting my completely (mostly) broken and dirty version of implementation of the phh-magisk-version, which I described in #1480 ?
And yes, you´re doing more, you build the latest version, but also creates the json file to implement it in the Magisk Manager.

**Comment by geecko86** [comment]
I just tried flashing the latest release of ExpressLuke/phh-magisk-builder.

I can boot (which is better than norma Magisk) but Magisk Manager tells me the version of Magisk that's on my device is unsupported (too old).

@PsyEng @soredake Do you guys have a suggestion on how to have a working Magisk install for an Android 11 GSI? 

I'm using a Xiaomi Mi Mix 3 with the latest releaes of phhusson/treble_experimentations - but I'm open to alternatives.

**Comment by PsyEng** [comment]
@geecko86 Maybe a little bit late, but use this [https://github.com/ExpressLuke/phh-magisk-builder/releases](https://github.com/ExpressLuke/phh-magisk-builder/releases)

**Comment by geecko86** [comment]
> @geecko86 Maybe a little bit late, but use this https://github.com/ExpressLuke/phh-magisk-builder/releases

@PsyEng I Flashed [Release 08-12-2020](https://github.com/ExpressLuke/phh-magisk-builder/releases/tag/08-12-2020) (Magisk 21.1) on top of Phh's [AOSP 11.0 v300.i](https://github.com/phhusson/treble_experimentations/releases/tag/v300.i) on a Xiaomi Mi Mix 3 (perseus).

I then installed the latest (canary) Magisk Manager. I'm still getting the "unsupported Magisk version" message.

What about you? Did you get it to work?

**Comment by geecko86** [comment]
Update: following these instructions made it work for me.

https://github.com/topjohnwu/Magisk/issues/3593#issuecomment-751534120

**Comment by dragonBall561** [comment]
> The recommended Magisk for this GSI https://github.com/phhusson/treble_experimentations/releases/download/v200/magisk-phh.zip
> Cannot be upgraded as it will cause a bootloop, also installing any other version other than that will also fail and cause bootloop. Is there any workaround for this?
> I'm using this GSI on my Moto Z2 Play https://sourceforge.net/projects/amber-gsi/files/PixelExperience_Plus/PixelExperience_Plus-10.0-arm64-a-gapps.img.xz/download
> With this KVT installer https://androidfilehost.com/?fid=8889791610682867863

Also I don't thing amber@gsi uses phh building scripts.
